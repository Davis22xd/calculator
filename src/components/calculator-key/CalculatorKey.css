.button {
    background-color: #fafafa;
    height: 36px;
    margin: 4px;
    width: 36px;
}