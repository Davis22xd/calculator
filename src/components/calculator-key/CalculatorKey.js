import React from "react";
import "./CalculatorKey.css";

export default class CalculatorKey extends React.Component {

  buttonClicked = () => {
    const { label, onbuttonClicked } = this.props;
    onbuttonClicked(label);
  }

  render() {
    const { label } = this.props;

    return (
      <button className="button" onClick={this.buttonClicked}>{label}</button>
    );
  }
}
