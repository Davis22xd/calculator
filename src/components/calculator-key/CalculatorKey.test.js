import * as React from "react";
import { shallow } from "enzyme";
import CalculatorKey from "./CalculatorKey";


describe("<CalculatorKey />", () => {
    const mockOnbuttonClicked = jest.fn();

    it("set a label and create a snapshot from it", () => {
      const props = {
        label: "+",
        onbuttonClicked: mockOnbuttonClicked
      };
      const wrapper = shallow(<CalculatorKey {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  