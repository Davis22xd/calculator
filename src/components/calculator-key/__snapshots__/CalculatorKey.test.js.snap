// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorKey /> set a label and create a snapshot from it 1`] = `ShallowWrapper {}`;
